opm/autodiff/BlackoilModelEbos.hpp

```diff
@@ -426,309 +381,209 @@ namespace Opm {
             typedef Dune::BCRSMatrix <MatrixBlockType>      Mat;
             typedef Dune::BlockVector<VectorBlockType>      BVector;
 
-            typedef Dune::MatrixAdapter<Mat,BVector,BVector> Operator;
             auto& ebosJac = ebosSimulator_.model().linearizer().matrix();
             auto& ebosResid = ebosSimulator_.model().linearizer().residual();
 
-            Operator opA(ebosJac);
+            typedef WellModelMatrixAdapter<Mat,BVector,BVector, ThisType> Operator;
+            Operator opA(ebosJac, const_cast< ThisType&  > (*this));
             const double relax = 0.9;
             typedef Dune::SeqILU0<Mat, BVector, BVector> SeqPreconditioner;
             SeqPreconditioner precond(opA.getmat(), relax);
             Dune::SeqScalarProduct<BVector> sp;
 
+            // apply well residual to the residual.
+            wellModel().apply(ebosResid);
```

Comment:
is the influence of the well model in the residual's Jacobian not considered? that might explain some of the convergence issues we were observing (some of those of the master branch, too).